frontend/src/pages/welcome.tsx

```diff
@@ -1,8 +1,291 @@
+        args: [""], // pass arguments
+      });
+
+      const hash = await writeContract(config, request);
+
+      await waitForTransactionReceipt(config, { hash });
+
+      alert("Registration successful");
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  // newMemberApprovalProposal()
+  const approveMember = async (e: MouseEvent) => {
+    e.preventDefault();
+    try {
+      const { request } = await simulateContract(config, {
+        abi: [], // Fill
+        address: "0x", // Fill
+        functionName: "newMemberApprovalProposal",
+        args: [""], // pass arguments
+      });
+
+      const hash = await writeContract(config, request);
+
+      await waitForTransactionReceipt(config, { hash });
+
+      alert("Member approved!");
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  // propose()
+  const propose = async (e: MouseEvent) => {
+    e.preventDefault();
+    try {
+      const { request } = await simulateContract(config, {
+        abi: [], // Fill
+        address: "0x", // Fill
+        functionName: "propose",
+        args: [""], // pass arguments
+      });
+
+      const hash = await writeContract(config, request);
+
+      await waitForTransactionReceipt(config, { hash });
+
+      alert("Members proposed!");
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  // castVote()
+  const castVote = async (e: MouseEvent) => {
+    e.preventDefault();
+    try {
+      const { request } = await simulateContract(config, {
+        abi: [], // Fill
+        address: "0x", // Fill
+        functionName: "castVote",
+        args: [""], // pass arguments
+      });
+
+      const hash = await writeContract(config, request);
+
+      await waitForTransactionReceipt(config, { hash });
+
+      alert("Vote casted!");
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  // executeProposal()
+  const executeProposal = async (e: MouseEvent) => {
+    e.preventDefault();
+    try {
+      const { request } = await simulateContract(config, {
+        abi: [], // Fill
+        address: "0x", // Fill
+        functionName: "executeProposal",
+        args: [""], // pass arguments
+      });
+
+      const hash = await writeContract(config, request);
+
+      await waitForTransactionReceipt(config, { hash });
+
+      alert("Proposal executed!");
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  // cancelProposal()
+  const cancelProposal = async (e: MouseEvent) => {
+    e.preventDefault();
+    try {
+      const { request } = await simulateContract(config, {
+        abi: [], // Fill
+        address: "0x", // Fill
+        functionName: "cancelProposal",
+        args: [""], // pass arguments
+      });
+
+      const hash = await writeContract(config, request);
+
+      await waitForTransactionReceipt(config, { hash });
+
+      alert("Proposal cancelled!");
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  // setVotingDelay()
+  const setDelay = async (e: MouseEvent) => {
```

Comment:
a state has already been declared with the same name at the top

Reply:
Resolved this one